proptest/src/array.rs

```diff
@@ -90,160 +107,111 @@ macro_rules! small_array {
         ///
         /// See [`UniformArrayStrategy`](struct.UniformArrayStrategy.html) for
         /// example usage.
-        pub fn $uni<S : Strategy>
-            (strategy: S) -> UniformArrayStrategy<S, [S::Value; $n]>
-        {
+        pub fn $uni<S: Strategy>(
+            strategy: S,
+        ) -> UniformArrayStrategy<S, [S::Value; $n]> {
             UniformArrayStrategy {
                 strategy,
-                _marker: PhantomData
+                _marker: PhantomData,
             }
         }
+    };
+}
 
-        impl<S : Strategy> Strategy for [S; $n] {
-            type Tree = ArrayValueTree<[S::Tree; $n]>;
-            type Value = [S::Value; $n];
-
-            fn new_tree(&self, runner: &mut TestRunner) -> NewTree<Self> {
-                Ok(ArrayValueTree {
-                    tree: [$(self[$ix].new_tree(runner)?,)*],
-                    shrinker: 0,
-                    last_shrinker: None,
-                })
-            }
-        }
+impl<S: Strategy, const N: usize> Strategy for [S; N] {
```

Comment:
do we want to add any test coverage for the const generic versions added?

Reply:
We already have a couple of tests that the manual impls used. Are you thinking specifically for the `uniform` function? Or just a test that it works with `[u8; 1000]` for example?

Reply:
just a little test that it works. i wouldn't block an approval on it but seems like a decent sanity check